src/test/java/com/microsoft/sqlserver/jdbc/NTLMConnectionTest.java

```diff
@@ -0,0 +1,536 @@
+        testInvalidProperties(
+                addOrOverrideProperty(connectionStringNTLM, "authentication", "ActiveDirectoryIntegrated "),
+                "R_SetAuthenticationWhenIntegratedSecurityTrue");
+    }
+
+    /**
+     * Tests NTLM authentication when the connection is encrypted.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testNTLMEncryptedConnection() throws SQLException {
+        try (Connection con = DriverManager
+                .getConnection(connectionStringNTLM + ";encrypt=true;trustServerCertificate=true")) {
+            verifyNTLM(con);
+        }
+    }
+
+    /**
+     * Tests NTLM authentication against a non-default database.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testNTLMNonDefaultDatabase() throws SQLException {
+        String databaseName = RandomUtil.getIdentifier("NTLM");
+        try (Statement stmt = connection.createStatement()) {
+            stmt.executeUpdate("CREATE DATABASE " + AbstractSQLGenerator.escapeIdentifier(databaseName));
+        }
+        try (Connection con = DriverManager
+                .getConnection(addOrOverrideProperty(connectionStringNTLM, "database", databaseName))) {
+            verifyNTLM(con);
+            verifyDatabase(con, databaseName);
+        }
+        try (Statement stmt = connection.createStatement()) {
+            TestUtils.dropDatabaseIfExists(databaseName, stmt);
+        }
+    }
+
+    /**
+     * Tests NTLM authentication when connection pooling is enabled.
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testNTLMHikariCP() throws SQLException {
+        HikariConfig config1 = new HikariConfig();
+        HikariConfig config2 = new HikariConfig();
+        config1.setDataSource(dsNTLMLocal);
+        config2.setJdbcUrl(connectionStringNTLM);
+        try (HikariDataSource dsCP1 = new HikariDataSource(config1);
+                HikariDataSource dsCP2 = new HikariDataSource(config2); Connection con1 = dsCP1.getConnection();
+                Connection con2 = dsCP2.getConnection()) {
+            verifyNTLM(con1);
+            verifyNTLM(con2);
+        }
+    }
+
+    /**
+     * TODO: random timeout failure with this test
+     * 
+     * Validates that loginTimeout connection property works with NTLM authentication.
+     */
+    // @Test
+    public void testNTLMLoginTimeout() {
+        long timerStart = 0;
+        long timerEnd = 0;
+        int loginTimeout = 10;
+
+        timerStart = System.currentTimeMillis();
+        try (Connection con = DriverManager
+                .getConnection(connectionStringNTLM + ";servername=bogus;instanceName=;loginTimeout=" + loginTimeout)) {
+            fail(TestResource.getResource("R_expectedFailPassed"));
+        } catch (SQLException e) {
+            timerEnd = System.currentTimeMillis();
+        }
+        long elapsedTime = (timerEnd - timerStart) / 1000;
+        assertTrue(elapsedTime < loginTimeout + 1);

```

Comment:
loginTimeout may not happen consistently (it would max +7 seconds) but not beyond that (Still you could allow 10 seconds relay). That's why I see the test is disabled. You may pass loginTimeout as 2 seconds and assert:

```suggestion
        assertTrue(elapsedTime < loginTimeout + 10);
```
Default timeout is 15 seconds in which case this would fail if applied timeout doesn't work.